presto-native-execution/presto_cpp/main/operators/LocalShuffle.h

```diff
@@ -28,24 +29,30 @@ namespace facebook::presto::operators {
 using TRowSize = uint32_t;
 constexpr size_t kUint32Size = sizeof(TRowSize);
 
+// Default buffer size for SortedShuffleFileStreamReader
+// This buffer is used for streaming reads from shuffle files during k-way
+// merge.
+constexpr size_t kDefaultStreamReaderBufferSize = 8 * 1024 * 1024; // 8MB
+
 // Metadata describing a serialized row's location and sizes in a buffer
 struct RowMetadata {
   size_t rowStart; // Offset to the start of this row
   uint32_t keySize; // Size of key (0 for non-sorted)
   uint32_t dataSize; // Size of data
 };
 
-// Compare two sort keys lexicographically
-inline bool compareKeys(std::string_view key1, std::string_view key2) noexcept {
-  return std::lexicographical_compare(
-      reinterpret_cast<const unsigned char*>(key1.data()),
-      reinterpret_cast<const unsigned char*>(key1.data() + key1.size()),
-      reinterpret_cast<const unsigned char*>(key2.data()),
-      reinterpret_cast<const unsigned char*>(key2.data() + key2.size()));
+// Compares sort keys lexicographically using three-way comparison.
+inline std::strong_ordering compareKeys(
```

Comment:
Not sure if we are fully migrated to C++20. Maybe for now let's keep the original bool